beprof/curve.py

```diff
@@ -95,15 +98,19 @@ def change_domain(self, domain):
         :param domain: set of points representing new domain. Might be a list or np.array
         :return: new Curve object with domain set by 'domain' parameter
         '''
-
+        logging.info('Running %s.change_domain() with new domain range:'
+                     ' [%s, %s]', self.__class__, np.min(domain), np.max(domain))
         # check if new domain includes in the orginal domain
-        if np.max(domain) > np.max(self.x):
-            # separate issue created to provide logging package
-            print('Error1')
-            return self
-        if np.min(domain) < np.min(self.x):
-            print('Error2')
+        try:
+            if np.max(domain) > np.max(self.x) or np.min(domain) < np.min(self.x):
+                raise IndexError
+        except IndexError as ie:
```

Comment:
It doesn't make sense to raise an exception and catch it immediately. Try to google: "Python exception best practices" and read a bit about them.